tests/ha/test_server_pod_failure.py

```diff
@@ -806,5 +806,391 @@ def test_degraded_deletes_safe_server_pod_shutdown(self):
+                    HA_CFG["common_params"]["60sec_delay"])
+        while len(rd_resp) != 4:
+            rd_resp = rd_output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        if not rd_resp:
+            assert_utils.assert_true(False, "Background process failed to do READs")
+        event_bkt_get = rd_resp[0]  # Contains buckets when event was set
+        fail_bkt_get = rd_resp[1]  # Contains buckets which failed when event was clear
+        event_di_bkt = rd_resp[2]  # Contains buckets when event was set
+        fail_di_bkt = rd_resp[3]  # Contains buckets which failed when event was clear
+
+        assert_utils.assert_false(len(fail_bkt_get) or len(fail_di_bkt),
+                                  "Expected PASS when event was clear, but buckets which failed in "
+                                  f"READs : {fail_bkt_get} OR DI check : {fail_di_bkt}")
+
+        LOGGER.info("Failed buckets while in-flight READs operation : %s", event_bkt_get)
+        LOGGER.info("Failed buckets while in-flight DI check operation : %s", event_di_bkt)
+
+        LOGGER.info("Step 6: Verifying responses from DELETEs background process")
+        del_resp = tuple()
+        LOGGER.info("Waiting for DELETE process output from Queue. Sleeping for %s",
+                    HA_CFG["common_params"]["60sec_delay"])
+        while len(del_resp) != 2:
+            del_resp = del_output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        if not del_resp:
+            assert_utils.assert_true(False, "Background process failed to do DELETEs")
+        event_del_bkt = del_resp[0]  # Contains buckets when event was set
+        fail_del_bkt = del_resp[1]  # Contains buckets which failed when event was clear
+        assert_utils.assert_false(len(fail_del_bkt), "Expected pass, buckets which failed in "
+                                                     f"DELETEs: {fail_del_bkt}.")
+        LOGGER.info("Failed buckets while in-flight DELETEs operation : %s", event_del_bkt)
+        LOGGER.info("Step 6: Verified responses from READs and DELETEs background processes")
+
+        remain_bkts = list(set(written_bck) - set(get_random_buck))
+        LOGGER.info(
+            "Step 7: Check READs-DI on remaining buckets of DELETEs operation: %s", remain_bkts)
+        new_s3data = {}
+        for bkt in remain_bkts:
+            new_s3data[bkt] = s3_data[bkt]
+
+        args = {'test_prefix': self.test_prefix, 'test_dir_path': self.test_dir_path,
+                'skipput': True, 'skipdel': True, 's3_data': new_s3data, 'di_check': True,
+                'output': rd_output}
+        self.ha_obj.put_get_delete(event, s3_test_obj, **args)
+        rd_resp = ()
+        while len(rd_resp) != 4:
+            rd_resp = rd_output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        if not rd_resp:
+            assert_utils.assert_true(False, "Failed to READ/Verify remaining buckets.")
```

Comment:
Here error message should be different. It may fail due to OS specific issue, and rd_resp will be empty.